services/net/contentmigration/ContentMigrationAction.cs

```diff
@@ -153,17 +161,35 @@ public class ContentMigrationAction : IngestAction<ContentMigrationOptions>
 
         DateTime? importDateStart = !string.IsNullOrEmpty(manager.Ingest.GetConfigurationValue("importDateStart")) ? manager.Ingest.GetConfigurationValue<DateTime>("importDateStart") : null;
         DateTime? importDateEnd = !string.IsNullOrEmpty(manager.Ingest.GetConfigurationValue("importDateEnd")) ? manager.Ingest.GetConfigurationValue<DateTime>("importDateEnd") : null;
-        DateTime? creationDateOfLastImport = !string.IsNullOrEmpty(manager.Ingest.GetConfigurationValue("creationDateOfLastImport")) ? manager.Ingest.GetConfigurationValue<DateTime>("creationDateOfLastImport") : null;
-        double migrationTimeOffsetInHours = !string.IsNullOrEmpty(manager.Ingest.GetConfigurationValue("migrationTimeOffsetInHours")) ? manager.Ingest.GetConfigurationValue<double>("migrationTimeOffsetInHours") : 0;
+        DateTime? creationDateOfLastImport = null;
+        try
+        {
+            creationDateOfLastImport = manager.Ingest.GetConfigurationValue<DateTime>("creationDateOfLastImport");
+        }
+        catch (Exception)
+        {
+            // migration has never been run before.
```

Comment:
Ignoring exceptions is always risky behaviour.  You should almost always log.

Reply:
sorted